core/Tracker/Visit.php

```diff
@@ -584,16 +586,18 @@ private function triggerPredicateHookOnDimensions($dimensions, $hook, Visitor $v
 
     protected function getAllVisitDimensions()
     {
-        $dimensions = VisitDimension::getAllDimensions();
+        if (is_null(self::$dimensions)) {
```

Comment:
At some point we can load this into `Piwik\Cache\Transient` but even transient cache is slow when called millions of times. Same applies to some other static caches that I used. We'll need to improve performance of transient cache at some point.